Mathlib/Tactic/Linter/DocModule.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+#  The "header" linter
+
+The "header" style linter checks that a file starts with
+```
+/-
+Copyright ...
+Apache ...
+Authors ...
+-/
+
+import*
+module doc-string*
+rest
+```
+It emits a warning if the first non-`import` command is not a module doc-string.
+-/
+
+open Lean Elab Command
+
+namespace Mathlib.Linter
+
+/-- `onlyImportsModDocs stx` checks whether `stx` is the syntax for a module that
+only consists of any number of `import` statements (possibly none) followed by
+either a module doc-string (and then anything else) or nothing else.
+-/
+def onlyImportsModDocs : Syntax → Option Bool
+  | .node _ ``Lean.Parser.Module.module #[_header, .node _ `null args] =>
+    let args := args
+    let first := args.get? 0
+    first.map (·.isOfKind ``Lean.Parser.Command.moduleDoc)
+  | _=> some false
+
+/-- returns the array of `import` identifiers. -/
+partial
+def getImportIds (s : Syntax) : Array Syntax :=
+  let rest : Array Syntax := (s.getArgs.map getImportIds).flatten
+  if s.isOfKind ``Lean.Parser.Module.import then
+    rest.push (s.getArgs.getD 2 default)
+  else
+    rest
+
+/-- `parseUpToHere stx post` takes as input a `Syntax` `stx` and a `String` `post`.
+It parses the file containing `stx` up to and excluding `stx`, appending `post` at the end.
+
+The option of appending a final string to the text gives more control to avoid syntax errors,
+for instance in the presence of `#guard_msgs in` or `set_option ... in`.
+-/
```

Comment:
It would be nice to put a warning here, how this will *not* suffice to parse every Lean file (as not all files can be parsed, but may need elaboration).

Reply:
Added a comment, will push soon!